clang/lib/Driver/Driver.cpp

```diff
@@ -846,16 +850,25 @@ static llvm::Triple getSYCLDeviceTriple(StringRef TargetArch) {
 
 static bool addSYCLDefaultTriple(Compilation &C,
                                  SmallVectorImpl<llvm::Triple> &SYCLTriples) {
+
+  llvm::Triple DefaultTriple = getSYCLDeviceTriple(
```

Comment:
@mdtoguchi Default triple will be `spirv32-unknown-unknown` or `spriv64-unknown-unknown`.
But at [L869](https://github.com/srividya-sundaram/llvm-project/pull/1/files#diff-f8c64e36dfe3828a6f816c4550e78bb0305769ace1be53207e86ac9a3280ac9eR869) , we check `SYCLTriple.isSPIROrSPIRV()` , which also checks if the arch is `spir` or `spir64`. 
Can it be just `SYCLTriple.isSPIRV()` instead, since the default triple we are constructing is only `spirv*` based arch.

Reply:
Yes, this makes sense for upstream since we are using spirv* values.  Downstream, this isn't guaranteed, so we need to be sure we keep an eye on this when it is pulled down.

Reply:
> Downstream, this isn't guaranteed, so we need to be sure we keep an eye on this when it is pulled down.

Can you elaborate this?

Reply:
In intel/llvm and xmain we support spir64, and still need to mess with the old model until we make the switch to the new model.